### PR TITLE
ci: simplify self-hosted runner selection

### PR DESCRIPTION
**Pull request by galargh**
This PR removes the need for extra job that was previously used to select runner type. It also makes the workflows that run on self-hosted runners, use runners that run the latest AMI (in particular, the AMI contains docker proxy setup and Go modules proxy setup). 

Finally, it moves some of the interop jobs to self-hosted because they were lagging behind others and making wait time for CI on PRs longer.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -44,7 +48,7 @@ jobs:
           path: cmd/ipfs/ipfs
   interop:
     needs: [interop-prep]
-    runs-on: ubuntu-latest
+    runs-on: ${{ fromJSON(github.repository == 'ipfs/kubo' && '["self-hosted", "linux", "x64", "2xlarge"]' || '"ubuntu-latest"') }}
```
Comment (guseggert) [inline .github/workflows/build.yml:51]: Just curious what ec2 instance type is "2xlarge" and "4xlarge" ?

Comment (galargh) [reply .github/workflows/build.yml:51]: These are from the C5 family in AWS - https://github.com/pl-strflt/tf-aws-gh-runner#runner-types

**Comment by galargh** [comment]
> > contains docker proxy setup and Go modules proxy setup
> 
> What does this mean?

We noticed that DockerHub was rate-limiting requests from our self-hosted runners. That's why we set up a read-through proxy to DockerHub. It caches docker layers on S3. This helps us prevent hitting limits and also speed up pulls.

Similarly, we hit some (unidentified yet) network issues with Go Modules Proxy. We dealt it with the same way, with a read-through proxy which caches stuff on S3.

**Review by hacdias — APPROVED** [review]

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -191,14 +195,26 @@ jobs:
...
         working-directory: ipfs-webui
-      - name: Run ipfs-webui@main build and smoke-test to confirm the upstream repo is not broken
-        run: npm test
+      - id: state
+        env:
+          GITHUB_REPOSITORY: ipfs/ipfs-webui
+          GITHUB_REF: ${{ steps.ref.outputs.ref }}
+          GITHUB_TOKEN: ${{ github.token }}
+        run: |
+          echo "state=$(curl -L -H "Authorization: Bearer $GITHUB_TOKEN" "https://api.github.com/repos/$GITHUB_REPOSITORY/commits/$GITHUB_REF/status" --jq '.state')" | tee -a $GITHUB_OUTPUT
```
Comment (lidel) [inline .github/workflows/build.yml:213]: @galargh  seems `curl` version at github does not (or no longer)  support `--jq`, check https://github.com/ipfs/kubo/actions/runs/5264574530/jobs/9516062618?pr=9940#step:10:17

Comment (galargh) [reply .github/workflows/build.yml:213]: I might have meant to use GH CLI here 🤦 Thanks for catching, I'll have it fixed.

**Pull request merged**